File: sample-banner/src/main/java/com/loopme/banner_sample/app/MainActivity.java
package com.loopme.banner_sample.app;

import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import com.loopme.LoopMeSdk;
import com.loopme.banner_sample.R;
import com.loopme.banner_sample.app.model.Constants;
import com.loopme.banner_sample.app.views.MainFeaturesFragment;
import com.loopme.banner_sample.app.views.RecyclerViewFragment;
import com.loopme.banner_sample.app.views.SimpleBannerFragment;

public class MainActivity
    extends AppCompatActivity
    implements MainFeaturesFragment.OnItemClickedListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_activity_layout);
        setAsMainView(MainFeaturesFragment.newInstance(), false);
        tryInitLoopMeSdk();
    }

    @Override
    public void onItemClicked(String item) {
        if (item.equalsIgnoreCase(Constants.SIMPLE)) {
            setAsMainView(SimpleBannerFragment.newInstance(), true);
        }
        if (item.equalsIgnoreCase(Constants.RECYCLERVIEW)) {
            setAsMainView(RecyclerViewFragment.newInstance(), true);
        }
        setTitle("View: " + item);
    }

    @Override
    public void onBackPressed() {
        Fragment fragment = getSupportFragmentManager().findFragmentById(R.id.main_root_view);
        if (fragment instanceof MainFeaturesFragment) {
            super.onBackPressed();
        } else {
            setAsMainView(MainFeaturesFragment.newInstance(), true);
            setTitle("LoopMe Banner Samples");
        }
    }

    private void setAsMainView(Fragment fragment, boolean replace) {
        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        if (replace) {
            fragmentTransaction.replace(R.id.main_root_view, fragment);
        } else {
            fragmentTransaction.add(R.id.main_root_view, fragment);
        }
        fragmentTransaction.commit();
    }

    private void alert(String message) {
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
    }

    private void tryInitLoopMeSdk() {
        if (LoopMeSdk.isInitialized()) return;
        alert("LoopMe SDK: initialization…");
        LoopMeSdk.initialize(this, new LoopMeSdk.Configuration(), new LoopMeSdk.LoopMeSdkListener() {
            @Override
            public void onSdkInitializationSuccess() {
                alert("LoopMe SDK: initialized");
            }
            @Override
            public void onSdkInitializationFail(int errorCode, String message) {
                alert("LoopMe SDK: failed to initialize. Trying again…");
                tryInitLoopMeSdk();
            }
        });
    }
}

File: sample-banner/src/main/java/com/loopme/banner_sample/app/MainActivity.kt
package com.loopme.banner_sample.app

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.loopme.LoopMeSdk
import com.loopme.LoopMeSdk.LoopMeSdkListener
import com.loopme.banner_sample.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        tryInitLoopMeSdk()
        setUpButtons()
    }

    private fun setUpButtons() {
        binding.banner.setOnClickListener {
            startActivity(
                Intent(
                    applicationContext,
                    BannerActivity::class.java
                )
            )
        }
        binding.interstitial.setOnClickListener {
            startActivity(
                Intent(
                    applicationContext,
                    InterstitialActivity::class.java
                )
            )
        }
    }

    private fun tryInitLoopMeSdk() {
        if (LoopMeSdk.isInitialized()) return
        alert("LoopMe SDK: initialization…")
        LoopMeSdk.initialize(this, LoopMeSdk.Configuration(), object : LoopMeSdkListener {
            override fun onSdkInitializationSuccess() {
                alert("LoopMe SDK: initialized")
            }

            override fun onSdkInitializationFail(errorCode: Int, message: String) {
                alert("LoopMe SDK: failed to initialize. Trying again…")
                tryInitLoopMeSdk()
            }
        })
    }

    private fun alert(message: String) {
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
    }
}